src/gc/env/gcenv.os.h

```diff
@@ -47,6 +47,42 @@ struct GCThreadAffinity
     int Processor;
 };
 
+// An event is a synchronization object whose state can be set and reset
+// indicating that an event has occured. It is used pervasively throughout
+// the GC.
+class GCEvent {
+public:
+    // Closes the event. Attempting to use the event past calling CloseEvent
+    // is a logic error.
+    virtual void CloseEvent() = 0;
```

Comment:
The GC-OS interface is statically linked, not virtualized. I do not think there is value in getting it virtualized. It is hard to tell where to stop once you start doing that, e.g. should new/delete operators be virtualized too? We have been there in desktop framework.

The different implementations of GCEvent won't co-exist. We should always need just one - so we should just include the right one to the build.